### PR TITLE
`Hierarchy.logiclId.path` is case sensitive

### PR DESCRIPTION
**Pull request by usefulalgorithm**
### 🤔 Why?

Because it should be case sensitive, but isn't.

### 🤓 What?

Fixes that.

### 🧪 Tested?

Unit test.

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: metaphor/snowflake/extractor.py [hunk 1]
```diff
@@ -399,13 +399,17 @@ def _add_system_tag(
         # Set hierarchy key and logical_id
         path = [DataPlatform.SNOWFLAKE.value]
         if object_type == "DATABASE":
-            path.extend([object_name])
+            path.extend(
+                [object_name.lower()]
+            )  # path should be normalized, i.e. should be lowercase
             hierarchy_key = dataset_normalized_name(object_name)
```
Comment (mars-lan) [inline metaphor/snowflake/extractor.py:405]: Path is supposed to contain the original casing, whereas `dataset.name` is noramlized to lower case.

**Review by mars-lan — CHANGES_REQUESTED** [review]

File: metaphor/snowflake/extractor.py (hunk 1 above)
Comment (mars-lan) [reply metaphor/snowflake/extractor.py:405]: Actually never mind. I got confused by the `structure` field. This is the correct fix.

Comment (mars-lan) [reply metaphor/snowflake/extractor.py:405]: On third thought. @alyiwang can you confirm if `logicalId.path` should be normalized?

Comment (usefulalgorithm) [reply metaphor/snowflake/extractor.py:405]: The problem is that `object_name` is in all caps

Comment (elic-eon) [reply metaphor/snowflake/extractor.py:405]: `structure` should be the only one that contains case-sensitive name

Comment (alyiwang) [reply metaphor/snowflake/extractor.py:405]: Yes, `path` is the ID of `hierarchy`, so should be normalized.

**Review by mars-lan — APPROVED** [review]

**Pull request merged**